packages/elasticsearch-plugin/src/build-elastic-body.ts

```diff
@@ -109,19 +108,34 @@ export function buildElasticBody(
             });
         }
         if (sort.price) {
-            const priceField = groupByProduct ? 'priceMin' : 'price';
+            // const priceField = groupByProduct ? 'priceMin' : 'price';
+            const priceField = 'price';
             sortArray.push({ [priceField]: { order: sort.price === SortOrder.ASC ? 'asc' : 'desc' } });
         }
     }
-    return {
-        query: searchConfig.mapQuery
-            ? searchConfig.mapQuery(query, input, searchConfig, channelId, enabledOnly)
-            : query,
-        sort: sortArray,
-        from: skip || 0,
-        size: take || 10,
-        track_total_hits: searchConfig.totalItemsMaxSize,
-    };
+
+    if (groupByProduct) {
+        return {
```

Comment:
These 2 `if` branches are almost identical, apart from the `collapse` property right? I think it makes sense to create a single object and then have add the `collapse` part after:
```ts
const body = { 
  // query, sort, from, size, track
};
if (groupByProduct) {
  body.collapse = { field: `productId` };
}
return body;
```